doc/Testing.md

```diff
@@ -67,6 +67,34 @@ For more documentation on testing, read the references listed at the bottom.
 	var cat = MockCat();
 	}
     ```
+
+- In some cases it's easier to mock with the use of the decorator pattern. First we decorate our repository like this :
+	```dart
+	class DadJokesRepositoryDecorator implements  DadJokesRepository{
+		final DadJokesRepository _innerRepository;
+		final MockingRepository _mockingRepository;
+
+		DadJokesRepositoryDecorator(this._innerRepository, this._mockingRepository);
+
+		@override
+		Future<DadJokeResponseData> getDadJokes() async {
+			var isMockingEnabled = await _mockingRepository.checkMockingEnabled();
+
+			if (isMockingEnabled) {
+				return mockedDadJokeResponse;
```

Comment:
I don't know if it's important, but the variable `mockedDadJokeResponse` doesn't seem to exist.

Reply:
it does exist it's located in dad_jokes_data_mock.dart